ignite/pkg/cosmosclient/cosmosclient.go

```diff
@@ -251,92 +309,65 @@ func (r Response) Decode(message proto.Message) error {
 	}, message)
 }
 
-// ConsensusInfo is the validator consensus info
-type ConsensusInfo struct {
-	Timestamp          string                `json:"Timestamp"`
-	Root               string                `json:"Root"`
-	NextValidatorsHash string                `json:"NextValidatorsHash"`
-	ValidatorSet       *tmproto.ValidatorSet `json:"ValidatorSet"`
+// Status returns the node status
+func (c Client) Status(ctx context.Context) (*ctypes.ResultStatus, error) {
+	return c.RPC.Status(ctx)
 }
 
-// ConsensusInfo returns the appropriate tendermint consensus state by given height
-// and the validator set for the next height
-func (c Client) ConsensusInfo(ctx context.Context, height int64) (ConsensusInfo, error) {
-	node, err := c.Context().GetNode()
+// BroadcastTx creates and broadcasts a tx with given messages for account.
+func (c Client) BroadcastTx(accountName string, msgs ...sdktypes.Msg) (Response, error) {
+	_, broadcast, err := c.BroadcastTxWithProvision(accountName, msgs...)
 	if err != nil {
-		return ConsensusInfo{}, err
+		return Response{}, err
 	}
+	return broadcast()
+}
 
-	commit, err := node.Commit(ctx, &height)
+func (c Client) GenerateTx(accountName string, msgs ...sdktypes.Msg) (string, error) {
```

Comment:
Yeah, good idea. I was indeed confused by that naming, so cleaning this up would be great. Will do!

Reply:
Then I think let's stick with the current API, option 1, where we get the accountName in CreateTx and calculate the gas beforehand. Otherwise the API will be a bit more complicated.

Reply:
@ilgooz I made the changes. Take a look at the design and see if it makes sense to you.